src/components/graph/util/model-functions.js

```diff
@@ -30,6 +30,16 @@ export function getFirstNodeOfType(elements, nodeType, buildStatus) {
     }
 }
 
+export function allowModificationsOnNode(workingNode, selectedNode) {
+    return (
+        !workingNode?.readOnly &&
+        workingNode?.buildStatus === 'BUILT' &&
+        (selectedNode === null ||
+            selectedNode === undefined ||
+            workingNode?.id === selectedNode?.id)
```

Comment:
This is a little too complex, maybe instead of one complex return, have multiple returns which are easier to read ?